File: ServerGuard.Agent/Config/AgentConfig.cs
using System.Text.Json.Serialization;

namespace ServerGuard.Agent.Config;
public class AgentConfig
{
    public Guid ResourceGroupId { get; set; }
    public Guid AgentId { get; set; }
    public bool IsCpuEnabled { get; set; }
    public bool IsGpuEnabled { get; set; }
    public bool IsMemoryEnabled { get; set; }
    public bool IsMotherboardEnabled { get; set; }
    public bool IsControllerEnabled { get; set; }
    public bool IsNetworkEnabled { get; set; }
    public bool IsStorageEnabled { get; set; }
    public int CollectEverySeconds { get; set; }
}

File: ServerGuard.Api/src/main/java/com/mmieczkowski/serverguard/model/AgentConfig.java
package com.mmieczkowski.serverguard.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

@Data
@Embeddable
public class AgentConfig implements Serializable {

    public AgentConfig(boolean isCpuEnabled,
                       boolean isGpuEnabled,
                       boolean isMemoryEnabled,
                       boolean isMotherboardEnabled,
                       boolean isControllerEnabled,
                       boolean isNetworkEnabled,
                       boolean isStorageEnabled,
                       int collectEverySeconds) {
        this.isCpuEnabled = isCpuEnabled;
        this.isGpuEnabled = isGpuEnabled;
        this.isMemoryEnabled = isMemoryEnabled;
        this.isMotherboardEnabled = isMotherboardEnabled;
        this.isControllerEnabled = isControllerEnabled;
        this.isNetworkEnabled = isNetworkEnabled;
        this.isStorageEnabled = isStorageEnabled;
        this.collectEverySeconds = collectEverySeconds;
    }

    public AgentConfig() {
    }

    @Column(nullable = false)
    private boolean isCpuEnabled;

    @Column(nullable = false)
    private boolean isGpuEnabled;

    @Column(nullable = false)
    private boolean isMemoryEnabled;

    @Column(nullable = false)
    private boolean isMotherboardEnabled;

    @Column(nullable = false)
    private boolean isControllerEnabled;

    @Column(nullable = false)
    private boolean isNetworkEnabled;

    @Column(nullable = false)
    private boolean isStorageEnabled;

    @Column(nullable = false, columnDefinition = "integer default 15")
    private int collectEverySeconds;

    @Column(nullable = true)
    private String apiKey;
}
